### PR TITLE
UIIN-503: Primary Contributor Layout Update

### PR DESCRIPTION
**Pull request by rasmuswoelk**
## Ticket
https://issues.folio.org/browse/UIIN-503

## Changes
- Replaced checkboxes with buttons for contributor fields when editing an instance
- Removed extra asterisk in contributor form
- Added "Primary" column for contributors when viewing an instance

May-06-2019 11-17-42

### CODE REVIEW

**Comment by rasmuswoelk** [comment]
@nielserik I accidentally left a `.only` in my tests 🙄It's removed now so it runs all tests again.

File: src/edit/contributorFields.js [hunk 1]
```diff
@@ -74,6 +68,35 @@ const ContributorFields = ({
...
+                  input.onChange(true);
+                };
+
+                return (
+                  <Fragment>
+                    { label && <Label>{label}</Label>}
+                    <Button
+                      data-test-edit-instance-contributors-primary-button
+                      buttonStyle={isPrimary ? 'primary' : 'default'}
+                      onClick={!isPrimary ? handleChange : null}
```
Comment (Yurii-Danylenko) [inline src/edit/contributorFields.js]: Is it possible to move `!isPrimary ? handleChange : null` to a handler method?

File: src/edit/contributorFields.js [hunk 2]
```diff
@@ -74,6 +68,35 @@ const ContributorFields = ({
               name: 'contributorTypeText',
               component: TextField,
             },
+            {
+              name: 'primary',
+              label: intl.formatMessage({ id: 'ui-inventory.primary' }),
+              component: ({ label, meta, input, fields }) => { /* eslint-disable-line react/prop-types */
```
Comment (Yurii-Danylenko) [inline src/edit/contributorFields.js]: What do you think about moving this field to a component?

Comment (rasmuswoelk) [reply src/edit/contributorFields.js]: I think that's a good idea. I'll move it into a new component 👍🏻

File: src/edit/contributorFields.js (hunk 1 above)
Comment (rasmuswoelk) [reply src/edit/contributorFields.js]: Sure 👍🏻

File: src/edit/components/PrimaryToggleButton/PrimaryToggleButton.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    onChange(true);
+  };
+
+  return (
+    <Fragment>
+      { label && <Label>{label}</Label>}
+      <Button
+        data-test-primary-toggle-button
+        buttonStyle={isPrimary ? 'primary' : 'default'}
+        onClick={handleChange}
```
Comment (Yurii-Danylenko) [inline src/edit/components/PrimaryToggleButton/PrimaryToggleButton.js]: I believe, `handleClick`, as a handler name, will look a little bit accurate. But it's up to you :)

Comment (rasmuswoelk) [reply src/edit/components/PrimaryToggleButton/PrimaryToggleButton.js]: Makes more sense, yes. I'll change it. Thanks :)

**Pull request merged**